### PR TITLE
Loan model created and added in inmemoryDB

### PR DESCRIPTION
**Pull request by bokimilinkovic**
https://trello.com/c/gvYHbh6B/78-create-loan-model-inmemory-data

### CODE REVIEW

File: domain/model/loan.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"github.com/google/uuid"
+	"time"
+)
+
+type State int
+const(
+	borrow State = 0
+	giveBack State = 1 // Not allowing me to call "return"
+)
```
Comment (sfqi) [inline domain/model/loan.go]: borrowed, returned

state, not State

even better, `type loanType int`

File: domain/model/loan.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+type Loan struct{
+	Id int
+	Transaction_id string
+	User_id int
+	Book_id int
+	Type State
+	Created_at time.Time
```
Comment (sfqi) [inline domain/model/loan.go]: ID, TransactionID, UserID, BookID, CreatedAt

File: domain/model/loan.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+func NewLoan() *Loan{
+	return &Loan{
+		Transaction_id: uuid.New().String(),
```
Comment (sfqi) [inline domain/model/loan.go]: `New()` panics if there's an error. We don't want to panic.

Might as well make it a proper constructor that takes three params and initializes the `Loan`.
Also, you could have `ReturnedLoan(...)` & `BorrowedLoan(...)` funcs that call `newLoan(...)`.

File: handler/handlers.go [hunk 4]
```diff
@@ -24,6 +24,18 @@ type store interface{
...
+	CreateUser(*model.User) error
+	UpdateUser(*model.User) error
+	FindAllUsers() ([]*model.User, error)
+	DeleteUser(*model.User)error
+
+	FindLoanById(int) (*model.Loan, error)
+	CreateLoan(*model.Loan) error
+	UpdateLoan(*model.Loan) error
+	FindAllLoans() ([]*model.Loan, error)
+	DeleteLoan(*model.Loan)error
```
Comment (sfqi) [inline handler/handlers.go]: Are you using these methods anywhere in the handlers? If not, why the interface?

File: repository/inmemory/loanDB.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package inmemory
+
+import (
+	"fmt"
+	"github.com/sfqi/library/domain/model"
+	"time"
+)
+
```
Comment (sfqi) [inline repository/inmemory/loanDB.go]: Now that I look at all this, I realize that the card (accurately) says "loan data", not "load methods" or whatever. We haven't defined what methods we'll be using. For one, there will be no deleting and updating loans.

File: repository/inmemory/loanDB.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	pointers := make([]*model.Loan, len(db.loans))
+	for i := 0; i < len(db.loans); i++ {
+		pointers[i] = &db.loans[i]
+	}
+	fmt.Println(pointers)
+	return pointers, nil
+}
+
+func (db *DB) CreateLoan(loan *model.Loan) error {
+	db.Id++
```
Comment (sfqi) [inline repository/inmemory/loanDB.go]: Pretty sure it shoudn't be `db.Id`

File: repository/inmemory/loanDB.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	now := time.Now()
+	loan.Created_at = now
+
+	loan.Id = db.Id
```
Comment (sfqi) [inline repository/inmemory/loanDB.go]:
```
loan := &{
    ID: db.<something>,
    CreatedAt: time.Now(),
}
```

File: repository/inmemory/loanDB.go [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+	return nil, -1, fmt.Errorf("error while finding Loan")
+}
+
+func (db *DB) FindAllLoans() ([]*model.Loan, error) {
+	pointers := make([]*model.Loan, len(db.loans))
+	for i := 0; i < len(db.loans); i++ {
+		pointers[i] = &db.loans[i]
+	}
+	fmt.Println(pointers)
+	return pointers, nil
```
Comment (sfqi) [inline repository/inmemory/loanDB.go]: Remind me why do we have to do this? Can't we just define `loans` as a slice of pointers to `model.Loan` objects?

Comment (bokimilinkovic) [reply repository/inmemory/loanDB.go]: We can, don't know how we miss this from the begging of the books creation, so i tried to be consistent..  I'll change this

File: repository/inmemory/loanDB.go (hunk 5 above)
Comment (bokimilinkovic) [reply repository/inmemory/loanDB.go]: Yeah i know, I assumed that we will be using our handler for operations over Loans ..  I misunderstood

File: domain/model/loan.go [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+	}
+}
+
+func ReturnedLoan(userid int, bookid int) *Loan{
+	loan := NewLoan()
+	loan.CreatedAt = time.Now()
+	loan.BookID = bookid
+	loan.UserID=userid
+	loan.Type = 1
+	return loan
```
Comment (sfqi) [inline domain/model/loan.go]:
```
func ReturnedLoad(...){
   return NewLoan(userID, bookID, returned)
}
```

File: handler/handlers.go [hunk 10]
```diff
@@ -24,6 +24,7 @@ type store interface{
 	UpdateBook(*model.Book) error
 	FindAllBooks() ([]*model.Book, error)
 	DeleteBook(*model.Book)error
+
```
Comment (sfqi) [inline handler/handlers.go]: No

**Pull request merged**